plugins/urls/__init__.py

```diff
@@ -1,165 +1,228 @@
-"""
-A plugin to provide various URL-related tools and services.
-
-This plugin does quite a lot of things - URL title parsing, URL shortening,
-URL logging to database, specialized URL handlers.. Its partner plugin,
-URL-tools (in the contrib repo) extends its functionality using the API it
-exposes.
-"""
+# coding=utf-8
+from copy import copy
+from twisted.python.failure import Failure
+from plugins.urls.shorteners.tinyurl import TinyURLShortener
+from system.protocols.generic.channel import Channel
 
 __author__ = 'Gareth Coles'
 
-import fnmatch
-import re
-import socket
-import urlparse
-import urllib
-import urllib2
-
-from bs4 import BeautifulSoup
+from collections import defaultdict
 from kitchen.text.converters import to_unicode
-from netaddr import all_matching_cidrs
+from twisted.internet.defer import Deferred, inlineCallbacks, returnValue
 
+# Managers
 from system.command_manager import CommandManager
-from system.decorators.threads import run_async_threadpool
 from system.event_manager import EventManager
-from system.events.general import MessageReceived
-
-import system.plugin as plugin
-
-from system.protocols.generic.channel import Channel
-from system.protocols.generic.user import User
-
-from system.storage.formats import YAML, DBAPI
 from system.storage.manager import StorageManager
 
-from .catcher import Catcher
+# Storage formats
+from system.storage.formats import Formats
 
-from system.translations import Translations
-_ = Translations().get()
-__ = Translations().get_m()
+# Plugin
+from system.plugins.plugin import PluginObject
 
+# Internals
+from plugins.urls.constants import PREFIX_TRANSLATIONS
+from plugins.urls.events import URLsPluginLoaded
+from plugins.urls.handlers.website import WebsiteHandler
+from plugins.urls.matching import extract_urls
+from plugins.urls.url import URL
 
-class URLsPlugin(plugin.PluginObject):
-    """
-    URLs plugin object
-    """
 
-    catcher = None
-    channels = None
+class URLsPlugin(PluginObject):
+    # Managers
     commands = None
-    config = None
     events = None
-    shortened = None
     storage = None
 
-    blacklist = []
-    handlers = {}
-    shorteners = {}
-    spoofing = {}
+    channels = None
+    config = None
+    shortened = None
 
-    content_types = ["text/html", "text/webviewhtml", "message/rfc822",
-                     "text/x-server-parsed-html", "application/xhtml+xml"]
+    shorteners = {}
```

Comment:
It's unlikely, but it's still an awful habit. Since any bugs caused by this will be subtle and manifest elsewhere, and is a one-line fix, it would be wise to change it.